### ISSUE TITLE
EXTENDED_TESTING env variable request during zef install? 

### ISSUE THREAD

**Issue by jubilatious1** [ugexe/zef#394]
## Context

It's not clear to me if this is a zef-related issue (more of a question than an issue). I successfully installed `db::pg` using the `--verbose` command-line flag (thank you for writing this module @CurtTilmes!). However all tests were skipped (1 author, 15 non-author) and for every one of 15 non-author tests the program warned:

`skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var`

## Expected Behavior

I imagine `zef` is sophisticated enough to manage testing protocols without resorting to `env` based (author-defined?) testing regimens.

## Actual Behavior

```
user@mbook:~$ ~/rakudo/rakudo-2020.10/zef/bin/zef install DB::Pg --verbose
===> Searching for: DB::Pg
===> Found: DB::Pg:ver<1.0>:auth<github:CurtTilmes>:api<1> [via Zef::Repository::LocalCache]
===> Testing: DB::Pg:ver<1.0>:auth<github:CurtTilmes>:api<1>
[DB::Pg] t/00-meta.t ............ skipped: test: To enable author tests, set AUTHOR_TESTING env var
[DB::Pg] t/01-basic.t ........... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/02-value.t ........... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/03-value-array.t ..... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/04-array.t ........... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/05-copy.t ............ skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/06-transaction.t ..... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/07-cursor.t .......... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/08-listen.t .......... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/09-native.t .......... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/10-type-converter.t .. skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/11-array-parser.t .... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/12-uuid.t ............ skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/13-datetime.t ........ skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/14-geometric.t ....... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] t/15-exception.t ....... skipped: test: To enable extended tests, set EXTENDED_TESTING or RELEASE_TESTING env var
[DB::Pg] Files=16, Tests=0,  4 wallclock secs ( 0.04 usr  0.03 sys +  4.44 cusr  0.78 csys =  5.29 CPU)
[DB::Pg] Result: NOTESTS
===> Testing [OK] for DB::Pg:ver<1.0>:auth<github:CurtTilmes>:api<1>
===> Installing: DB::Pg:ver<1.0>:auth<github:CurtTilmes>:api<1>
===> Install [OK] for DB::Pg:ver<1.0>:auth<github:CurtTilmes>:api<1>
user@mbook:~$ 

```

## Your Environment

I suppose I don't have the requested `env` variables set (either `AUTHOR_TESTING` or `EXTENDED_TESTING`). Note, it would be lovely to try the `--force-tests` command line flag after the fact, but `zef` warns:
 ` All candidates are currently installed`
`No reason to proceed. Use --force-install to continue anyway`

```
user@mbook:~$ raku -v
Welcome to 𝐑𝐚𝐤𝐮𝐝𝐨™ v2020.10.
Implementing the 𝐑𝐚𝐤𝐮™ programming language v6.d.
Built on MoarVM version 2020.10.
user@mbook:~$ ~/rakudo/rakudo-2020.10/zef/bin/zef --version
v0.10.0
```

**Comment by tony-o** [comment]
Usually module authors for this sort of thing have an environment set up that would likely fail under less controlled circumstances. It’s a common thing to use either a different directory for those tests or to use environment variables when you need information in the test (like a db username). 

If you think your env is setup for author tests then set the environment vars and use `-f` to force reinstall.

**Comment by jubilatious1** [comment]
Thanks @tony-o . I've looked for explanations on how to do `env` -based testing in Raku and have come up empty handed. Specifically I've found what is referred to as the 'Lancaster Consensus' for Perl5, but nothing for Raku:

https://github.com/Perl-Toolchain-Gang/toolchain-site/blob/master/lancaster-consensus.md#environment-variables-for-testing-contexts

I also just want to make a minor point that 'extended' testing is a misnomer (above). The module installed with no tests being run whatsoever. This might be a problem down-the-road with spoofed author addresses, etc.

My inclination is for you or @ugexe to close this issue and I'll open a new one either on https://github.com/Raku/ecosystem OR https://github.com/Raku/problem-solving

Best Regards.
